x/ibc-rate-limit/ibc_module.go

```diff
@@ -134,15 +135,22 @@ func (im *IBCModule) OnRecvPacket(
 		return channeltypes.NewErrorAcknowledgement("bad packet in rate limit's OnRecvPacket")
 	}
 
-	channelValue := im.ics4Middleware.CalculateChannelValue(ctx, denom, packet)
+	channelValue := im.ics4Middleware.CalculateChannelValue(ctx, denom)
+
+	channel := packet.GetSourceChannel()
+	if !strings.HasPrefix(denom, "ibc/") {
+		// For native channels, we want to have only one limit across all channels. We can use a quota for
+		//channel "any" for this
+		channel = "any"
+	}
```

Comment:
Shouldn't this be for all denoms now?

Reply:
I thought we wanted to do this only for native tokens. IBC tokens are mostly limited to a channel anyway, but we could end up in the case where someone "parks" the tokens on another network through a different channel. 

If we want the all tokens to have an "any" channel, then we can probably remove the channel abstraction altogether (though it might make sense to keep it in case we want to add it back in the future)

Reply:
I've moved this logic to the contract. Now the actual channel and "any" are checked independently. This way we can get one behaviour or the other depending on how we configure the quotas. Marking as resolved.